File: Algoritmos-em-C/binary-search.c
#include <stdio.h>
#define TARGET 76

int binarySearch(int* array, int size, int target){
    int low = 0;
    int high = size-1;
    while(low <= high){
        int middle = (low + high)/2;
        int guess = array[middle];
        if(guess == target){
            return middle;
        }
        if(guess > target){
            high = middle-1;
        } else {
            low = middle+1;
        }
    }
    return -1;
}

int main(){
    int array[] = {5, 10, 25, 53, 68, 72, 76, 81, 89, 97};
    int size = sizeof(array)/sizeof(array[0]);
    printf("%d\n\n", binarySearch(array, size, TARGET));
    
    return 0;
}

File: Algoritmos-em-Python/binary-search.py
def binary_search(array, target) -> int:
    low = 0
    high = len(array) -1
    while(low <= high):
        middle = int((low + high)/2)
        guess = array[middle]
        if guess == target:
            return middle;
        if guess > target:
            high = middle-1
        else:
            low = middle+1  

    return None


sample_array = [1, 2, 5, 10, 25, 31, 32, 57, 63, 99, 115, 225, 300, 543]

print(binary_search(sample_array, 300))